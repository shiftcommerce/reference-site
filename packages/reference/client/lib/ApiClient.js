// Libraries
import qs from 'qs'
import axios from 'axios'

class ApiClient {
  constructor (options = {}) {
    this.client = axios.create({
      baseURL: process.env.API_HOST_PROXY
    })
  }

  async read (endpoint, queryObject = {}, options = {}) {
    const formattedEndpoint = this.encodeParams(endpoint, queryObject)

    try {
      const response = await this.client.get(formattedEndpoint)
      return { status: response.status, data: response.data }
    } catch (error) {
      console.error('API CLIENT: Error while fetching data', error)
      return { status: error.response.status, data: error.response.data }
    }
  }

  async post (endpoint, body = {}, options = {}) {
    try {
      const response = await this.client.post(endpoint, body)
      return { status: response.status, data: response.data }
    } catch (error) {
      console.error('Error while posting data', error)
      return { status: error.response.status, data: error.response.data }
    }
  }

  encodeParams (endpoint, queryObject = {}) {
    if (Object.keys(queryObject).length > 0) {
      const query = qs.stringify(queryObject)
      endpoint = `${endpoint}?${query}`
    }
    return endpoint
  }
}

export default ApiClient
