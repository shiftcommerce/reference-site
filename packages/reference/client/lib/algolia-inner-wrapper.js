// Lib
import InitialPropsDelegator from './initial-props-delegator'
import getConfig from 'next/config'
import { Configure } from 'react-instantsearch/dom'

// Components
import { InstantSearch } from '../components/search/algolia/instant-search'

const {
  publicRuntimeConfig: {
    ALGOLIA_API_KEY,
    ALGOLIA_APP_ID,
    ALGOLIA_INDEX_NAME
  }
} = getConfig()

export default function algoliaInnerWrapper (Component) {
  return class extends InitialPropsDelegator(Component) {
    render () {
      const { resultsState, onSearchStateChange, searchState, ...otherProps } = this.props
      return (
        <InstantSearch
          appId={ALGOLIA_APP_ID}
          apiKey={ALGOLIA_API_KEY}
          indexName={ALGOLIA_INDEX_NAME}
          resultsState={resultsState}
          onSearchStateChange={onSearchStateChange}
          searchState={searchState}
        >
          <Configure {...searchState.configure} />
          <Component {...otherProps} />
        </InstantSearch>
      )
    }
  }
}
