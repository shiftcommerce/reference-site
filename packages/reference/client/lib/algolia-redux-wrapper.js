// Libraries
import withReduxStore from './with-redux-store'
import { Provider } from 'react-redux'
import { Configure } from 'react-instantsearch/dom'
import Router from 'next/router'
import qs from 'qs'
import getConfig from 'next/config'

// Lib
import InitialPropsDelegator from './initial-props-delegator'
import buildSearchStateForURL from '../lib/build-search-state-for-url'

// Components
import ConnectedLayout from '../components/layout'
import { InstantSearch, findResultsState } from '../components/search/algolia/instant-search'

const {
  publicRuntimeConfig: {
    ALGOLIA_API_KEY,
    ALGOLIA_APP_ID,
    ALGOLIA_INDEX_NAME,
    ALGOLIA_RESULTS_PER_PAGE
  }
} = getConfig()

export default function (ConnectedComponent, Page) {
  return algoliaOuterWrapper(
    withReduxStore(
      withProvider(
        algoliaInnerWrapper(
          withLayout(
            ConnectedComponent
          )
        )
      )
    ),
    Page
  )
}

function algoliaOuterWrapper (NextWrapper, Page) {
  return class extends InitialPropsDelegator(NextWrapper) {
    constructor (props) {
      super(props)
      this.onSearchStateChange = this.onSearchStateChange.bind(this)
    }

    static async getInitialProps (appContext) {
      const { query } = appContext

      // Get initial props from the parent
      const appProps = await Object.getPrototypeOf(this).getInitialProps(appContext)

      let searchState = query
      if (Page.buildAlgoliaStates) {
        searchState = Page.buildAlgoliaStates(appContext)
      }

      let resultsState = {
        _originalResponse: {
          results: [{}]
        }
      }
      if (Page.algoliaEnabled && Page.algoliaEnabled()) {
        resultsState = await findResultsState(NextWrapper, { searchState, ...appProps })
      }

      return {
        ...appProps,
        resultsState,
        searchState
      }
    }

    updateAfter () {
      // Delegate to custom implementation if the page provides one
      if (Page.updateAfter) {
        return Page.updateAfter()
      }

      // Default implementation
      return 700
    }

    onSearchStateChange (searchState) {
      // Delegate to custom implementation if the page provides one
      if (Page.onSearchStateChange) {
        return Page.onSearchStateChange.call(this, searchState)
      }

      // Default implementation
      clearTimeout(this.debouncedSetState)
      this.debouncedSetState = setTimeout(() => {
        const href = this.searchStateToUrl(searchState)
        Router.push(href, href, { shallow: true })
      }, this.updateAfter())
      this.setState({ searchState })
    }

    searchStateToUrl (searchState) {
      // Delegate to custom implementation if the page provides one
      if (Page.searchStateToUrl) {
        return Page.searchStateToUrl.call(this, searchState)
      }

      // Default implementation
      const urlSearchState = buildSearchStateForURL(searchState)
      // Build the query string and append it to search path
      return searchState.query ? `/search?${qs.stringify(urlSearchState)}` : '/search'
    }

    componentDidMount () {
      return Page.algoliaComponentDidMount && Page.algoliaComponentDidMount.call(this)
    }

    componentWillReceiveProps (nextProps) {
      return Page.algoliaComponentWillReceiveProps && Page.algoliaComponentWillReceiveProps.call(this, nextProps)
    }

    render () {
      const { resultsState, searchState, ...appProps } = this.props
      return (
        <NextWrapper
          resultsState={resultsState}
          onSearchStateChange={this.onSearchStateChange}
          searchState={
            this.state && this.state.searchState
              ? this.state.searchState
              : searchState
          }
          {...appProps}
        />
      )
    }
  }
}

function withProvider (Component) {
  return class extends InitialPropsDelegator(Component) {
    render () {
      const { reduxStore, ...otherProps } = this.props
      return (
        <Provider store={reduxStore}>
          <Component {...otherProps} />
        </Provider>
      )
    }
  }
}

function algoliaInnerWrapper (Component) {
  return class extends InitialPropsDelegator(Component) {
    render () {
      const { resultsState, onSearchStateChange, searchState, ...otherProps } = this.props
      // We should always configure this in the state to prevent a onSearchStateChange first render
      searchState.configure = { ...searchState.configure, hitsPerPage: ALGOLIA_RESULTS_PER_PAGE }
      return (
        <InstantSearch
          appId={ALGOLIA_APP_ID}
          apiKey={ALGOLIA_API_KEY}
          indexName={ALGOLIA_INDEX_NAME}
          resultsState={resultsState}
          onSearchStateChange={onSearchStateChange}
          searchState={searchState}
        >
          <Configure {...searchState.configure} />
          <Component searchState={ searchState } {...otherProps} />
        </InstantSearch>
      )
    }
  }
}

function withLayout (Component) {
  return class extends InitialPropsDelegator(Component) {
    render () {
      return (
        <ConnectedLayout>
          <Component {...this.props} />
        </ConnectedLayout>
      )
    }
  }
}
