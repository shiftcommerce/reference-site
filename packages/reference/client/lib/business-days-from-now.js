// Libraries
import moment from 'moment-business-days'

moment.locale('en-gb')

export function businessDaysFromNow (days) {
  return moment(new Date()).businessAdd(days)
}
