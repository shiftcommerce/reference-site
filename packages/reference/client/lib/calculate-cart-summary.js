export function calculateCartSummary (cart, checkout = {}) {
  const lineItems = cart.lineItems
  const shippingMethod = checkout.shippingMethod || {}
  const shipping = shippingMethod.retail_price_inc_tax || 0

  let [ total, discount, subTotal ] = [ 0, 0, 0 ]

  lineItems.map((li, index) => {
    subTotal += (li.quantity * li.price)
    discount += (li.quantity * (li.discount || 0))
  })
  total = subTotal - discount + shipping

  const totals = {
    subTotal: subTotal,
    total: total,
    discount: discount,
    shipping: shipping
  }
  return totals
}
