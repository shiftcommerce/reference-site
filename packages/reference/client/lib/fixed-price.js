export function fixedPrice (price) {
  return (price).toFixed(2)
}
