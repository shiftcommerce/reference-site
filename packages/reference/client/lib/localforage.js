import localforage from 'localforage'

class LocalForage {
  constructor () {
    if (typeof window !== 'undefined') {
      this.initLocalForage()
    }
  }

  // Set the driver to indexedDB
  initLocalForage () {
    localforage.config({
      driver: localforage.INDEXEDDB,
      name: 'CART'
    })
  }

  // Set the key and value in the indexedDB
  setValue (key, value) {
    localforage.setItem(key, value).then((value) => {
    }).catch((err) => {
      console.log('Error in setting key value:', err)
    })
  }

  // Get the value from the indexedDB
  async getValue (key, callback) {
    let keyValue = await localforage.getItem(key).then((value) => {
      return value
    }).catch((err) => {
      console.log('Error in retrieving the key:', err)
    })
    return keyValue
  }
}
export default LocalForage
