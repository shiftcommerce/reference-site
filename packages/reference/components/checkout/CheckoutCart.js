// Libraries
import { Component } from 'react'

// Objects
import Image from '../../objects/Image'
import Link from 'next/link'

export class CheckoutCart extends Component {
  renderLineItems () {
    return (
      this.props.cart.lineItems.map((lineItem, index) =>
        <div aria-label='Product' key={index} className='o-checkout-cart__product'>
          <div className='o-checkout-cart__image'>
            <Image src={lineItem.imageUrl} alt={lineItem.title} width={80} aria-label={lineItem.title} />
          </div>

          <ul className='o-checkout-cart__info'>
            <li aria-label='Product title'>{ lineItem.title }</li>
            <li aria-label='Quantity'>Qty: { lineItem.quantity }</li>
            <li aria-label='Price'>Price: &pound;{ lineItem.price }</li>
            <li aria-label='Subtotal'>Subtotal: &pound;{ lineItem.price * lineItem.quantity }</li>
          </ul>
        </div>
      )
    )
  }

  render () {
    const { title } = this.props
    return (
      <div aria-label='Cart items summary' className='o-checkout-cart'>
        <div className='o-checkout-cart__header'>
          <div>
            <h2>{ title }</h2>
          </div>

          <div>
            <Link href='/cart'>
              <a aria-label='Edit your cart' className='o-button o-button--lrg'>Edit Cart</a>
            </Link>
          </div>
        </div>

        <div className='o-checkout-cart__wrapper'>
          { this.renderLineItems() }
        </div>
      </div>
    )
  }
}

export default CheckoutCart
