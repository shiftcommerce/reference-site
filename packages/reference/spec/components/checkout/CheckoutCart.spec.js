// Components
import CheckoutCart from '../../../components/checkout/CheckoutCart'

// Lib 
import { fixedPrice } from '../../../lib/fixedPrice'

describe('Checkout Cart', () => {
  test('renders the correct cart summary with single line item', () => {
    // arrange
    const cart = {
      lineItems: [
        {
          title: 'Test Product',
          price: fixedPrice(10.0),
          discount: 0,
          quantity: 2,
          sku: '123',
          imageUrl: '',
          size: 'size - 8',
          stockAvailableLevel: '1000',
          variant: '38 Waist 31 Leg',
          slug: '1',
          canonical_path: '1'
        }
      ]
    }

    // act
    const wrapper = mount(
      <CheckoutCart cart={cart} />
    )

    // assert
    expect(wrapper).toMatchSnapshot()
    expect(wrapper.find('h4.c-line-items__details-title')).toIncludeText('Test Product')
    expect(wrapper.find('div.c-line-items__total-wrapper')).toIncludeText('Â£20')
  })

  test('renders the correct cart summary with multiple line items', () => {
    // arrange
    const cart = {
      lineItems: [
        {
          title: 'Test Product',
          price: fixedPrice(10.0),
          discount: 0,
          quantity: 2,
          stockAvailableLevel: '1000',
          variant: '38 Waist 31 Leg',
          slug: '1',
          canonical_path: '1'
        },
        {
          title: 'Pretend Product',
          price: fixedPrice(5.0),
          discount: 0,
          quantity: 1,
          stockAvailableLevel: '1000',
          variant: '38 Waist 31 Leg',
          slug: '1',
          canonical_path: '1'
        }
      ]
    }

    // act
    const wrapper = mount(
      <CheckoutCart cart={cart} />
    )

    // assert
    expect(wrapper).toMatchSnapshot()
    expect(wrapper.find('div.c-line-items')).toIncludeText('Test Product')
    expect(wrapper.find('div.c-line-items')).toIncludeText('Pretend Product')
  })
})
