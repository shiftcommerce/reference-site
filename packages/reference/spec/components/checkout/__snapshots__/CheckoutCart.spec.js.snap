// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkout Cart renders the correct cart summary with multiple line items 1`] = `
<CheckoutCart
  cart={
    Object {
      "lineItems": Array [
        Object {
          "canonical_path": "1",
          "discount": 0,
          "price": "10.00",
          "quantity": 2,
          "slug": "1",
          "stockAvailableLevel": "1000",
          "title": "Test Product",
          "variant": "38 Waist 31 Leg",
        },
        Object {
          "canonical_path": "1",
          "discount": 0,
          "price": "5.00",
          "quantity": 1,
          "slug": "1",
          "stockAvailableLevel": "1000",
          "title": "Pretend Product",
          "variant": "38 Waist 31 Leg",
        },
      ],
    }
  }
>
  <section
    className="c-cart-table__header"
  >
    <div
      className="u-float--left"
    >
      <h2
        className="c-cart-table__title"
      >
         Your Shopping Basket 
        <a
          className="c-cart-table__amount"
        >
          (
          )
        </a>
         
      </h2>
      <p
        className="c-cart-table__description"
      >
        You have 
        <a />
         
        <t
          className={null}
          count={1}
          showCount={false}
          singular="item"
          style={Object {}}
          zero={null}
        >
          <span
            className={null}
            style={Object {}}
          >
            item
          </span>
        </t>
         in your shopping basket
      </p>
      <p
        className="c-cart-table__description"
      >
        <t
          className={null}
          count={1}
          plural="These"
          showCount={false}
          singular="This"
          style={Object {}}
          zero={null}
        >
          <span
            className={null}
            style={Object {}}
          >
            This
          </span>
        </t>
         
        <t
          className={null}
          count={1}
          showCount={false}
          singular="item"
          style={Object {}}
          zero={null}
        >
          <span
            className={null}
            style={Object {}}
          >
            item
          </span>
        </t>
         will be saved for 48 hours depending on availablility
      </p>
    </div>
    <div
      className="u-float--right"
    >
      <h4
        className="c-cart-table__total"
      >
        £
        25.00
      </h4>
    </div>
  </section>
  <LineItems
    cart={
      Object {
        "lineItems": Array [
          Object {
            "canonical_path": "1",
            "discount": 0,
            "price": "10.00",
            "quantity": 2,
            "slug": "1",
            "stockAvailableLevel": "1000",
            "title": "Test Product",
            "variant": "38 Waist 31 Leg",
          },
          Object {
            "canonical_path": "1",
            "discount": 0,
            "price": "5.00",
            "quantity": 1,
            "slug": "1",
            "stockAvailableLevel": "1000",
            "title": "Pretend Product",
            "variant": "38 Waist 31 Leg",
          },
        ],
      }
    }
  >
    <div
      className="c-line-items"
    >
      <div
        className="o-grid-container__lineitems"
        key="1"
      >
        <div
          className="o-grid-item1__lineitems"
        >
          <Link
            as="1"
            href="/slug?slug=1"
            key="1"
          >
            <Image
              alt="Test Product"
              aria-label="Test Product"
              className="c-line-items__image"
              key="1"
              onClick={[Function]}
            >
              <div
                className="c-dummy_image c-line-items__image"
                style={
                  Object {
                    "height": undefined,
                    "width": undefined,
                  }
                }
              />
            </Image>
          </Link>
        </div>
        <div
          className="o-grid-item2__lineitems"
        >
          <div
            className="c-line-items__details"
          >
            <h4
              className="c-line-items__details-title  u-bold"
            >
              Test Product
            </h4>
            <div
              className="c-line-items__details-sku"
            >
              <span />
            </div>
          </div>
        </div>
        <div
          className="o-grid-item3__lineitems"
        >
          <div
            className="c-line-items__colour  u-bold"
          >
            <span>
              Colour
            </span>
          </div>
          <div
            className="c-line-items__size  u-bold"
          >
            <span>
              Option
            </span>
          </div>
          <div
            className="c-line-items__quantity u-bold"
          >
            <span>
              Quantity
            </span>
          </div>
        </div>
        <div
          className="o-grid-item4__lineitems"
        >
          <span
            className="c-line-items__space"
          >
            Grey
          </span>
          <span
            className="c-line-items__space"
          >
            38 Waist 31 Leg
          </span>
          <span
            className="c-line-items__space"
          >
            <select
              value={2}
            >
              <option
                aria-posinset={2}
                aria-setsize={1}
                key="1"
                value={1}
              >
                1
              </option>
              <option
                aria-posinset={3}
                aria-setsize={2}
                key="2"
                value={2}
              >
                2
              </option>
              <option
                aria-posinset={4}
                aria-setsize={3}
                key="3"
                value={3}
              >
                3
              </option>
              <option
                aria-posinset={5}
                aria-setsize={4}
                key="4"
                value={4}
              >
                4
              </option>
              <option
                aria-posinset={6}
                aria-setsize={5}
                key="5"
                value={5}
              >
                5
              </option>
              <option
                aria-posinset={7}
                aria-setsize={6}
                key="6"
                value={6}
              >
                6
              </option>
              <option
                aria-posinset={8}
                aria-setsize={7}
                key="7"
                value={7}
              >
                7
              </option>
              <option
                aria-posinset={9}
                aria-setsize={8}
                key="8"
                value={8}
              >
                8
              </option>
              <option
                aria-posinset={10}
                aria-setsize={9}
                key="9"
                value={9}
              >
                9
              </option>
              <option
                aria-posinset={11}
                aria-setsize={10}
                key="10"
                value={10}
              >
                10
              </option>
            </select>
          </span>
        </div>
        <div
          className="o-grid-item5__lineitems"
        >
          <div
            className="c-line-items__buttons"
          >
            <div
              className="c-line-items__anchor"
            >
              <a
                className="c-line-items__delete"
              >
                Delete
              </a>
            </div>
          </div>
          <div
            className="c-line-items__total-wrapper"
          >
            <a
              className="c-line-items__total"
            >
              £
              20.00
            </a>
          </div>
        </div>
      </div>
      <div
        className="o-grid-container__lineitems"
        key="1"
      >
        <div
          className="o-grid-item1__lineitems"
        >
          <Link
            as="1"
            href="/slug?slug=1"
            key="1"
          >
            <Image
              alt="Pretend Product"
              aria-label="Pretend Product"
              className="c-line-items__image"
              key="1"
              onClick={[Function]}
            >
              <div
                className="c-dummy_image c-line-items__image"
                style={
                  Object {
                    "height": undefined,
                    "width": undefined,
                  }
                }
              />
            </Image>
          </Link>
        </div>
        <div
          className="o-grid-item2__lineitems"
        >
          <div
            className="c-line-items__details"
          >
            <h4
              className="c-line-items__details-title  u-bold"
            >
              Pretend Product
            </h4>
            <div
              className="c-line-items__details-sku"
            >
              <span />
            </div>
          </div>
        </div>
        <div
          className="o-grid-item3__lineitems"
        >
          <div
            className="c-line-items__colour  u-bold"
          >
            <span>
              Colour
            </span>
          </div>
          <div
            className="c-line-items__size  u-bold"
          >
            <span>
              Option
            </span>
          </div>
          <div
            className="c-line-items__quantity u-bold"
          >
            <span>
              Quantity
            </span>
          </div>
        </div>
        <div
          className="o-grid-item4__lineitems"
        >
          <span
            className="c-line-items__space"
          >
            Grey
          </span>
          <span
            className="c-line-items__space"
          >
            38 Waist 31 Leg
          </span>
          <span
            className="c-line-items__space"
          >
            <select
              value={1}
            >
              <option
                aria-posinset={2}
                aria-setsize={1}
                key="1"
                value={1}
              >
                1
              </option>
              <option
                aria-posinset={3}
                aria-setsize={2}
                key="2"
                value={2}
              >
                2
              </option>
              <option
                aria-posinset={4}
                aria-setsize={3}
                key="3"
                value={3}
              >
                3
              </option>
              <option
                aria-posinset={5}
                aria-setsize={4}
                key="4"
                value={4}
              >
                4
              </option>
              <option
                aria-posinset={6}
                aria-setsize={5}
                key="5"
                value={5}
              >
                5
              </option>
              <option
                aria-posinset={7}
                aria-setsize={6}
                key="6"
                value={6}
              >
                6
              </option>
              <option
                aria-posinset={8}
                aria-setsize={7}
                key="7"
                value={7}
              >
                7
              </option>
              <option
                aria-posinset={9}
                aria-setsize={8}
                key="8"
                value={8}
              >
                8
              </option>
              <option
                aria-posinset={10}
                aria-setsize={9}
                key="9"
                value={9}
              >
                9
              </option>
              <option
                aria-posinset={11}
                aria-setsize={10}
                key="10"
                value={10}
              >
                10
              </option>
            </select>
          </span>
        </div>
        <div
          className="o-grid-item5__lineitems"
        >
          <div
            className="c-line-items__buttons"
          >
            <div
              className="c-line-items__anchor"
            >
              <a
                className="c-line-items__delete"
              >
                Delete
              </a>
            </div>
          </div>
          <div
            className="c-line-items__total-wrapper"
          >
            <a
              className="c-line-items__total"
            >
              £
              5.00
            </a>
          </div>
        </div>
      </div>
    </div>
  </LineItems>
</CheckoutCart>
`;

exports[`Checkout Cart renders the correct cart summary with single line item 1`] = `
<CheckoutCart
  cart={
    Object {
      "lineItems": Array [
        Object {
          "canonical_path": "1",
          "discount": 0,
          "imageUrl": "",
          "price": "10.00",
          "quantity": 2,
          "size": "size - 8",
          "sku": "123",
          "slug": "1",
          "stockAvailableLevel": "1000",
          "title": "Test Product",
          "variant": "38 Waist 31 Leg",
        },
      ],
    }
  }
>
  <section
    className="c-cart-table__header"
  >
    <div
      className="u-float--left"
    >
      <h2
        className="c-cart-table__title"
      >
         Your Shopping Basket 
        <a
          className="c-cart-table__amount"
        >
          (
          )
        </a>
         
      </h2>
      <p
        className="c-cart-table__description"
      >
        You have 
        <a />
         
        <t
          className={null}
          count={1}
          showCount={false}
          singular="item"
          style={Object {}}
          zero={null}
        >
          <span
            className={null}
            style={Object {}}
          >
            item
          </span>
        </t>
         in your shopping basket
      </p>
      <p
        className="c-cart-table__description"
      >
        <t
          className={null}
          count={1}
          plural="These"
          showCount={false}
          singular="This"
          style={Object {}}
          zero={null}
        >
          <span
            className={null}
            style={Object {}}
          >
            This
          </span>
        </t>
         
        <t
          className={null}
          count={1}
          showCount={false}
          singular="item"
          style={Object {}}
          zero={null}
        >
          <span
            className={null}
            style={Object {}}
          >
            item
          </span>
        </t>
         will be saved for 48 hours depending on availablility
      </p>
    </div>
    <div
      className="u-float--right"
    >
      <h4
        className="c-cart-table__total"
      >
        £
        20.00
      </h4>
    </div>
  </section>
  <LineItems
    cart={
      Object {
        "lineItems": Array [
          Object {
            "canonical_path": "1",
            "discount": 0,
            "imageUrl": "",
            "price": "10.00",
            "quantity": 2,
            "size": "size - 8",
            "sku": "123",
            "slug": "1",
            "stockAvailableLevel": "1000",
            "title": "Test Product",
            "variant": "38 Waist 31 Leg",
          },
        ],
      }
    }
  >
    <div
      className="c-line-items"
    >
      <div
        className="o-grid-container__lineitems"
        key="1"
      >
        <div
          className="o-grid-item1__lineitems"
        >
          <Link
            as="1"
            href="/slug?slug=1"
            key="1"
          >
            <Image
              alt="Test Product"
              aria-label="Test Product"
              className="c-line-items__image"
              key="1"
              onClick={[Function]}
              src=""
            >
              <div
                className="c-dummy_image c-line-items__image"
                style={
                  Object {
                    "height": undefined,
                    "width": undefined,
                  }
                }
              />
            </Image>
          </Link>
        </div>
        <div
          className="o-grid-item2__lineitems"
        >
          <div
            className="c-line-items__details"
          >
            <h4
              className="c-line-items__details-title  u-bold"
            >
              Test Product
            </h4>
            <div
              className="c-line-items__details-sku"
            >
              <span>
                123
              </span>
            </div>
          </div>
        </div>
        <div
          className="o-grid-item3__lineitems"
        >
          <div
            className="c-line-items__colour  u-bold"
          >
            <span>
              Colour
            </span>
          </div>
          <div
            className="c-line-items__size  u-bold"
          >
            <span>
              Option
            </span>
          </div>
          <div
            className="c-line-items__quantity u-bold"
          >
            <span>
              Quantity
            </span>
          </div>
        </div>
        <div
          className="o-grid-item4__lineitems"
        >
          <span
            className="c-line-items__space"
          >
            Grey
          </span>
          <span
            className="c-line-items__space"
          >
            38 Waist 31 Leg
          </span>
          <span
            className="c-line-items__space"
          >
            <select
              data-variant="123"
              value={2}
            >
              <option
                aria-posinset={2}
                aria-setsize={1}
                key="1"
                value={1}
              >
                1
              </option>
              <option
                aria-posinset={3}
                aria-setsize={2}
                key="2"
                value={2}
              >
                2
              </option>
              <option
                aria-posinset={4}
                aria-setsize={3}
                key="3"
                value={3}
              >
                3
              </option>
              <option
                aria-posinset={5}
                aria-setsize={4}
                key="4"
                value={4}
              >
                4
              </option>
              <option
                aria-posinset={6}
                aria-setsize={5}
                key="5"
                value={5}
              >
                5
              </option>
              <option
                aria-posinset={7}
                aria-setsize={6}
                key="6"
                value={6}
              >
                6
              </option>
              <option
                aria-posinset={8}
                aria-setsize={7}
                key="7"
                value={7}
              >
                7
              </option>
              <option
                aria-posinset={9}
                aria-setsize={8}
                key="8"
                value={8}
              >
                8
              </option>
              <option
                aria-posinset={10}
                aria-setsize={9}
                key="9"
                value={9}
              >
                9
              </option>
              <option
                aria-posinset={11}
                aria-setsize={10}
                key="10"
                value={10}
              >
                10
              </option>
            </select>
          </span>
        </div>
        <div
          className="o-grid-item5__lineitems"
        >
          <div
            className="c-line-items__buttons"
          >
            <div
              className="c-line-items__anchor"
            >
              <a
                className="c-line-items__delete"
                data-variant="123"
              >
                Delete
              </a>
            </div>
          </div>
          <div
            className="c-line-items__total-wrapper"
          >
            <a
              className="c-line-items__total"
            >
              £
              20.00
            </a>
          </div>
        </div>
      </div>
    </div>
  </LineItems>
</CheckoutCart>
`;
